espflash/src/chip/mod.rs

```diff
@@ -69,7 +69,9 @@ pub trait ChipType: ReadEFuse {
         Ok(bytes_to_mac_addr(bytes))
     }
 
-    fn supports_target(target: &str) -> bool;
+    fn supports_target(target: &str) -> bool {
```

Comment:
I intentionally made the `supports_target` check loser than the `SUPPORTED_TARGETS` list to allow using customized/future targets for the device that `espflash` doesn't know about.

Reply:
Okay that's fair, thank you for giving your rationale.

My main concern with this approach is that you are able to provide any arbitrary number of targets assuming they have the correct prefix, regardless of whether or not these targets exists. This bypasses any error handling currently done by `miette` and will result in `rustc` throwing an error instead. To me, this defeats the purpose of even checking the target in the first place.

I'm not sure that custom targets will really be a common issue, and since we control the Rust compiler fork we will know of any new targets being added ahead of time and can act accordingly.

If you feel strongly about this then I will revert these changes, though.